src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -462,14 +461,14 @@ void setData(byte[] data) {
         }
 
         NatsMessage getMessage() {
-            NatsMessage message;
-            if (replyTo != null && replyTo.startsWith("$JS")) {
-                message = new NatsJetStreamMessage();
-            }
-            else if (status != null) {
+            NatsMessage message = null;
+            if (status != null) {
                 message = new StatusMessage();
             }
-            else {
+            else if (JsPrefixManager.isJsMessage(replyTo)) {

```

Comment:
nitpick... `hasPrefix` might be a better name.   Up to you...

Reply:
changed